Support screenshot image filenames containing parentheses

peruukki opened the pull request:
I have a website that has a page with parentheses in its URL. When running e.g. `wraith capture`, cropping the page’s screenshots and generating thumbnails fails with errors like these:

```
cropping images
sh: -c: line 0: syntax error near unexpected token `('
sh: -c: line 0: `convert shots/__programme__tutt(i)uus/1024__auraco.png -background none -extent 1024x3801 shots/__programme__tutt(i)uus/1024__auraco.png'
cropping images
sh: -c: line 0: syntax error near unexpected token `('
sh: -c: line 0: `convert shots/__programme__tutt(i)uus/320__auraco.png -background none -extent 320x4202 shots/__programme__tutt(i)uus/320__auraco.png'
...
Generating thumbnails
sh: -c: line 0: syntax error near unexpected token `('
sh: -c: line 0: `convert shots/__programme__tutt(i)uus/1024__auraco.png -thumbnail 200 -crop 200x200+0+0 shots/thumbnails/__programme__tutt(i)uus/1024__auraco.png'
sh: -c: line 0: syntax error near unexpected token `('
sh: -c: line 0: `convert shots/__programme__tutt(i)uus/320__auraco.png -thumbnail 200 -crop 200x200+0+0 shots/thumbnails/__programme__tutt(i)uus/320__auraco.png'
```

I solved this by escaping the screenshot filenames before passing them to shell execution. I started writing separate tests for this case too, but in the end I felt it resulted in too much (almost duplicate) test code. So I just test this using the existing tests with an adjusted filename.

peruukki commented:
This also seems to fix #202.

bootstraponline commented on lib/wraith/compare_images.rb:
```diff
@@ -27,7 +28,8 @@ def percentage(img_size, px_value, info)
 
   def compare_task(base, compare, output, info)
-    cmdline = "compare -dissimilarity-threshold 1 -fuzz #{wraith.fuzz} -metric AE -highlight-color blue #{base} #{compare} #{output}"
+    compare_escaped = Shellwords.escape(compare)
```
I think you can just do compare.shellescape

bootstraponline commented:
shellwords needs to be added to the gemspec as a runtime dependency.

bootstraponline commented:
> This pull request contains merge conflicts that must be resolved.
> Only those with write access to this repository can merge pull requests.

and the pull request needs to be rebased

peruukki commented:
@bootstraponline, thanks for the tip about calling `shellescape` directly on the string, it makes the changes much nicer one-liners! Shellwords is part of the Ruby standard library, so apparently it shouldn't be added to runtime depedencies; I tried adding it, but then the gem installation failed to this:
```
ERROR:  While executing gem ... (Gem::UnsatisfiableDependencyError)
    Unable to resolve dependency: 'wraith (= 2.4.1)' requires 'shellwords (>= 0)'
```
I changed the commit to call `shellescape` everywhere and rebased it.

bootstraponline commented:
> Shellwords is part of the Ruby standard library, so apparently it shouldn't be added to runtime depedencies;

oh, sorry didn't know that. The PR looks good to me. :+1:

bootstraponline commented:
/cc @DaveBlooman
